orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java

```diff
@@ -890,6 +890,57 @@ public void testUseMatchQueryOnEmbeddedNumericIdCreatesTermQuery() throws Except
 		);
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-1793")
+	public void testQueryingForEntitiesFromDifferentIdSpaces() throws Exception {
+		QueryBuilder queryBuilder = fullTextSession.getSearchFactory()
+				.buildQueryBuilder().forEntity( EducationalInstitution.class ).get();
+
+		Query query = queryBuilder.bool()
```

Comment:
Personally I would love to have a more direct test, which for example just tests `CriteriaObjectInitializer` and that for example x amount of `Criteria` queries get created/executed. The contract of  `CriteriaObjectInitializer` is of course hard to test stand-alone. Maybe one could do something with a Byteman test!?